Data Retention Policy API Docs (#556)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/organizations.mdx

## Show Module Producers

<EnterpriseAlert>

> This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.

Reviewer: ```suggestion
This endpoint is exclusive to Terraform Enterprise and is not available in Terraform Cloud.
```
We need to make this update for all enterprise alert badges.

}
```

> ## Show Data Retention Policy

Author: @trujillo-adam I know that `BEGIN: TFC:only` works, but I don't see anywhere else that uses `BEGIN: TFE:only`, and it seems like it [didn't exist previously](https://hashicorp.slack.com/archives/CSHKFQC4A/p1677522271974239?thread_ts=1677085125.348179&cid=CSHKFQC4A). Should we be using it now?

<EnterpriseAlert>
This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.
</EnterpriseAlert>

`GET /organizations/:organization_name/relationships/data-retention-policy`

| Parameter            | Description                                                         |
| ---------------------| --------------------------------------------------------------------|
| `:organization_name` | The name of the organization to show the data retention policy for. |

This endpoint shows the data retention policy set on the organization.

> When no data retention policy is set on the organization, the site wide data retention policy will be used by default. Read more about [organization data retention policies](/terraform/cloud-docs/users-teams-organizations/organizations#data-retention-policies).

Author: yes, good point on the link. I was using `cloud-docs` all over the place so that things would work in the Vercel preview, but I can start moving content over to ptfe-releases

The endpoint can also be used to change the type of the data retention policy on the organization. Making a request with the new type of policy will automatically create a data retention policy of the new type and replace the existing organization policy. For more information on the types of data retention policies, refer to [Data Retention Policy Types](/terraform/cloud-docs/api-docs/data-retention-policies#types-of-data-retention-policies).

| Parameter            | Description                                                                |
| -------------------- | -------------------------------------------------------------------------- |
| `:organization_name` | The name of the organization to update the data retention policy for.      |

| Status  | Response                  | Reason(s)                                                      |
| ------- | ------------------------- | -------------------------------------------------------------- |
| [204][] | No Content                | Successfully updated the organization data retention policy.   |

> | [404][] | [JSON API error object][] | Organization not found, or user unauthorized to perform action.|

Reviewer 2: Non-Blocking Nitpick: Should we add a space here between the period and the pipe?:
```suggestion
| [404][] | [JSON API error object][] | Organization not found, or user unauthorized to perform action. |
```

### Request Body

> This POST endpoint requires a JSON object with the following properties as a request payload.

Reviewer 2: Suggestion: Should we format this for consistency?:
```suggestion
This `POST` endpoint requires a JSON object with the following properties as a request payload.
```

| `data.type`       | string |         | A [data retention policy type](/terraform/cloud-docs/api-docs/data-retention-policies#types-of-data-retention-policies) |
| `data.attributes` | object |         | The attributes for the [chosen policy type](/terraform/cloud-docs/api-docs/data-retention-policies#types-of-data-retention-policies) |

### Sample Payload

```json
{
  "data": {
    "type": "data-retention-policy-delete-olders",
    "attributes": {
      "deleteOlderThanNDays": 33
    }
  }
}
```

### Sample Request

```shell
$ curl \
  --header "Authorization: Bearer $TOKEN" \
  --header "Content-Type: application/vnd.api+json" \
  --request POST \
  --data @payload.json \
  https://app.terraform.io/api/v2/organizations/hashicorp/relationships/data-retention-policy
```

### Sample Response

```json
{
  "data": {
    "id": "drp-Kh74zpKVGNWbpugQ",
    "type": "data-retention-policy-delete-olders",
    "attributes": {
      "delete-older-than-n-days": 33
    },
    "relationships": {
      "target": {
        "data": {
          "id": "hashicorp",
          "type": "organization"
        }
      }
    }
  }
}
```

> ## Remove data retention policy

Reviewer 2: Non-Blocking Nitpick (Same as above!): Should we titlecase this for consistency between the other headers and this one?:
```suggestion
## Remove Data Retention Policy
```

<EnterpriseAlert>
This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.
</EnterpriseAlert>

`DELETE /organizations/:organization_name/relationships/data-retention-policy`

| Parameter            | Description                                                                                                                                                                          |
| ---------------------| -------------------------------------------------------------------------- |
| `:organization_name` | The name of the organization to remove the data retention policy for. |

This endpoint removes the data retention policy explicitly set on an organization.

> Organizations without a data retention policy will default to using the site-wide data retention policy when determining how long to retain backing data.

Reviewer 2: Non-Blocking Suggestion: Would it be beneficial to link the site admin docs here for easy access?:
```suggestion
Organizations without a data retention policy will default to using the [site-wide data retention policy](https://developer.hashicorp.com/terraform/enterprise/application-administration/general#data-retention-policies) when determining how long to retain backing data.
```

}
```

## Show data retention policy

<EnterpriseAlert>

> This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.

Reviewer: ```suggestion
This endpoint is exclusive to Terraform Enterprise and is not available in Terraform Cloud.
```

</EnterpriseAlert>

`GET /organizations/:organization_name/relationships/data-retention-policy`

| Parameter            | Description                                                         |
| ---------------------| --------------------------------------------------------------------|
| `:organization_name` | The name of the organization to show the data retention policy for. |

This endpoint shows the data retention policy set explicitly on the organization.

When no data retention policy is set for the organization, the endpoint returns the default policy configured for the Terraform Enterprise installation. Read more about [organization data retention policies](/terraform/enterprise/users-teams-organizations/organizations#data-retention-policies).

> For additional information, refer to [Data Retention Policy Types](/terraform/enterprise/api-docs/data-retention-policies#data-retention-policy-types).

Reviewer: ```suggestion
For additional information, refer to [Data Retention Policy Types](/terraform/enterprise/api-docs/data-retention-policies#data-retention-policy-types) in the Terraform Enterprise documentation.
```
Just to make it clear that the link takes readers to a different set. Please make this change for other links from TFC to TFE.

## Create or update data retention policy

<EnterpriseAlert>

> This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.

Reviewer: ```suggestion
This endpoint is exclusive to Terraform Enterprise and is not available in Terraform Cloud.
```

This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.
</EnterpriseAlert>

`POST /organizations/:organization_name/relationships/data-retention-policy`

| Parameter            | Description                                                                |
| -------------------- | -------------------------------------------------------------------------- |
| `:organization_name` | The name of the organization to update the data retention policy for.      |

This endpoint creates a data retention policy for an organization, or updates the existing policy.

> Read more about [organization data retention policies](/terraform/cloud-docs/users-teams-organizations/organizations#data-retention-policies).

Reviewer 2: Non-blocking nitpick: Should we add an extra line between this line of text and the previous line of text to keep it consistent with the [text on line 948](https://github.com/hashicorp/terraform-docs-common/pull/556/files#diff-f5625a4be7f4b34305dd570e3a8eb331c34e7fd7bc58dab8f5499ed46ceb8737R948)?:
```suggestion

Read more about [organization data retention policies](/terraform/cloud-docs/users-teams-organizations/organizations#data-retention-policies).
```

See [Data Retention Policy API](/terraform/enterprise/api-docs/data-retention-policies#create-or-update-data-retention-policy) for details.

## Remove data retention policy

<EnterpriseAlert>

> This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.

Reviewer: ```suggestion
This endpoint is exclusive to Terraform Enterprise and is not available in Terraform Cloud.
```

</EnterpriseAlert>

`DELETE /organizations/:organization_name/relationships/data-retention-policy`

| Parameter            | Description                                                                                                                                                                          |
| ---------------------| -------------------------------------------------------------------------- |
| `:organization_name` | The name of the organization to remove the data retention policy for. |

This endpoint removes the data retention policy explicitly set on an organization.
When the data retention policy is deleted, the organization inherits the default policy configured for the Terraform Enterprise installation. Refer to [Data Retention Policies](/terraform/enterprise/application-administration/general#data-retention-policies) for additional information.

Refer to [Data Retention Policies](/terraform/enterprise/users-teams-organizations/organizations#data-retention-policies) for information about configuring data retention policies for an organization.

> See [Data Retention Policy API](/terraform/enterprise/api-docs/data-retention-policies#remove-data-retention-policy) for details.

Reviewer: ```suggestion
Refer to [Data Retention Policy API](/terraform/enterprise/api-docs/data-retention-policies#remove-data-retention-policy) for details.
```
We're trying to standardize on "Refer to" instead of "See".
